src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -106,6 +106,9 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             marshaller.marshal(toSave, fileWriter);
 
         } catch (IOException ioe) {
+            if (!Files.isWritable(path)){
```

Comment:
No need to check for the writability explicitly. Exception already gives you that information. 
All you need to do is to handle it!